File: tcfs15p/cubemod.cpp
#include<iostream>
using namespace std;
int pow(int a,int b);
int cubemod(int a,int b,int c);
int main(){
	int t;
	cin>>t;
	int i,a,b,c;
	for(i=0;i<t;i++){
		cin>>a>>b>>c;
		cout<<cubemod(a,b,c)<<endl;;
	}
}
int pow(int a,int b){
	int i=1,j=0;
	for(j=0;j<b;j++){
		i=(i*a);
	}
	return i;
}
int cubemod(int a,int b,int c){
	int roots=0,i;
	for(i=a;i<=b;i++){
		if((((i*(i+1)+1)%c)==0))
			roots++;
	}
	return roots;
}

File: tcfs15p/cubemod.py
import numpy as np;
import sys;
t=int(raw_input());

def cubemod(a):
	roots=0;
#	print a,a[0],a[1],a[2];
	for i in xrange(a[0],a[1]+1):
		if(( ( (i**3) - 1) % a[2])==0):
			roots+=1;
	return roots;				

def main():
	for i in xrange(t):
		a=raw_input().split();
		a=map(int,a);
		print cubemod(a);
main();